src/utils/artifact/generateAragonArtifact.ts

```diff
@@ -0,0 +1,106 @@
+import { AragonAppJson, AragonArtifact, AragonEnvironment } from '~/src/types'
+import { keccak256, AbiItem } from 'web3-utils'
+import { getAppId } from '../appName'
+import { parseContractFunctions, AragonContractFunction } from '../ast'
+import { keyBy } from 'lodash'
+import { ethers } from 'ethers'
+
+const abiFallback = {
+  payable: true,
+  stateMutability: 'payable',
+  type: 'fallback'
+}
+
+function _generateAragonArtifact(
+  arapp: AragonAppJson,
+  abi: AbiItem[],
+  functions: AragonContractFunction[]
+): AragonArtifact {
+  const abiFunctions = abi.filter(abiElem => abiElem.type === 'function')
+  const abiBySignature = keyBy(abiFunctions, ethers.utils.formatSignature)
+
+  // Get mainnet env
+  const mainnetEnv: AragonEnvironment | undefined =
+    arapp.environments['mainnet']
+  const appName = (mainnetEnv || {}).appName || ''
+
```

Comment:
This is no longer relevant if we want to enforce the same ipfs hash across environments.

Reply:
Added in https://github.com/aragon/buidler-aragon/issues/77